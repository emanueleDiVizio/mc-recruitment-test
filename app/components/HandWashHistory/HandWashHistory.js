import React from 'react'
import { useSelector } from 'react-redux'
import { View, FlatList, Text } from 'react-native'

const handWashHistorySelector = state => state.handWashes

const HandWashItem = ({ endTime }) => {
  return (
    <View>
      <Text>Handwash completed on {new Date(endTime).toString()}</Text>
    </View>
  )
}

const HandWashHistory = () => {
  const data = useSelector(handWashHistorySelector)
  return (
    <View>
      <FlatList
        data={data}
        renderItem={({ item }) => <HandWashItem endTime={item.endTime} />}
        keyExtractor={item => item.endTime.toString()}
      />
    </View>
  )
}

export default HandWashHistory
