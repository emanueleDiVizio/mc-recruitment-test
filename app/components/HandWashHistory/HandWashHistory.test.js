import React from 'react'
import HandWashHistory from './HandWashHistory'
import { render } from '@testing-library/react-native'
import * as redux from 'react-redux'

describe('HandWashComponent', () => {
  let state

  beforeEach(() => {
    redux.useSelector = selector => selector(state)
  })

  it('renders correctly fetching data from store', () => {
    state = {
      handWashes: [
        { endTime: 1589532292430 },
        { endTime: 1589532322793 },
        { endTime: 1589532333570 },
      ],
    }
    const { baseElement } = render(<HandWashHistory />)

    expect(baseElement).toMatchSnapshot()
  })
})
