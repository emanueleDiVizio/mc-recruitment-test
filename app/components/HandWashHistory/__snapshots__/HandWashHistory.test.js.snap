// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandWashComponent renders correctly fetching data from store 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#4A4A4A",
            "marginBottom": 8,
            "marginLeft": 16,
          }
        }
      >
        HISTORY
      </Text>
      <ScrollView
        data={
          Array [
            Object {
              "endTime": 1589532292430,
            },
            Object {
              "endTime": 1589532322793,
            },
            Object {
              "endTime": 1589532333570,
            },
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        maxToRenderPerBatch={10}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "margin": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#00A778",
                  "justifyContent": "center",
                  "marginRight": 8,
                  "width": 70,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "padding": 8,
                  }
                }
              >
                08:44
              </Text>
            </View>
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                Friday
              </Text>
              <Text
                style={
                  Object {
                    "color": "#9B9B9B",
                    "fontSize": 14,
                  }
                }
              >
                15th May 2020
              </Text>
            </View>
          </View>
        </View>
        <View
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "margin": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#00A778",
                  "justifyContent": "center",
                  "marginRight": 8,
                  "width": 70,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "padding": 8,
                  }
                }
              >
                08:45
              </Text>
            </View>
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                Friday
              </Text>
              <Text
                style={
                  Object {
                    "color": "#9B9B9B",
                    "fontSize": 14,
                  }
                }
              >
                15th May 2020
              </Text>
            </View>
          </View>
        </View>
        <View
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "margin": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#00A778",
                  "justifyContent": "center",
                  "marginRight": 8,
                  "width": 70,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "padding": 8,
                  }
                }
              >
                08:45
              </Text>
            </View>
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                Friday
              </Text>
              <Text
                style={
                  Object {
                    "color": "#9B9B9B",
                    "fontSize": 14,
                  }
                }
              >
                15th May 2020
              </Text>
            </View>
          </View>
        </View>
      </ScrollView>
    </View>
  </View>
</View>
`;
